### ISSUE TITLE
[native-image] JNI/LLVM failure with ClassCastException

### ISSUE THREAD

**Issue by johanvos** [oracle/graal#1622]
When compiling HelloWorld using LLVM backend, a number of ClassCastExceptions are thrown.
Command:
```
native-image \
-H:CompilerBackend=llvm \
-H:-AOTInline \
-H:-SpawnIsolates \
-Dsvm.platform="org.graalvm.nativeimage.impl.InternalPlatform\$DARWIN_JNI_AMD64" \
-cp build/classes/java/main hello.HelloWorld

```
exception:

```
java.lang.ClassCastException: class com.oracle.svm.jni.access.JNIAccessibleMethod cannot be cast to class com.oracle.svm.hosted.code.CEntryPointData (com.oracle.svm.jni.access.JNIAccessibleMethod and com.oracle.svm.hosted.code.CEntryPointData are in unnamed module of loader 'app')
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMGenerator.<init>(SubstrateLLVMGenerator.java:97)
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMBackend.newLLVMGenerator(SubstrateLLVMBackend.java:119)
        at org.graalvm.compiler.core.llvm.LLVMCompilerBackend.emitLLVM(LLVMCompilerBackend.java:61)
        at org.graalvm.compiler.core.llvm.LLVMCompilerBackend.emitBackEnd(LLVMCompilerBackend.java:42)
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMBackend.emitBackEnd(SubstrateLLVMBackend.java:113)
```

I added some debug, and for example this method causes a ClassCastException:
`HostedMethod<JNIJavaCallWrappers.jniInvoke_ARRAY:Ljava_lang_Boolean_2_0002e_0003cinit_0003e_00028Z_00029V -> AnalysisMethod<JNIJavaCallWrappers.jniInvoke_ARRAY:Ljava_lang_Boolean_2_0002e_0003cinit_0003e_00028Z_00029V -> com.oracle.svm.jni.hosted.JNIJavaCallWrapperMethod@2297a625>>`

**Comment by pejovica** [comment]
I reproduced the failure, I'm working on a fix.

**Comment by pejovica** [comment]
Fixed in 40a1a80d4b67fd2ee0ccb9725d5f368895983316.
